### DISCUSSION TITLE
I have a question about the file editor

### DISCUSSION THREAD

**Discussion by AlbertDijksma** [HASwitchPlate/openHASP#248] [Q&A] [answered]
Hi, I have a question about the file editor.
I am missing a save button to save the changes.
Is this correct?
My setup:
Lolin TFT 2.4" Touch Shield
Lilygo Ttgo T7 V1.5 Mini32 ESP32-WROVER-B Psram
Software version master from 6-11-2021

**Comment by nagyrobi** [comment, marked as answer]
Ctrl+S

**Reply by fvanroie** [reply]
I have added this question to the FAQ

**Comment by samturner3** [comment]
How do we persist the file changes between reboots?
I am able to save the file via above, and when I come back into the file editor it is there; however after reboot its gone.

**Reply by nagyrobi** [reply]
CTRL+S will save permanently, and that survives reboots. If it doesn't, there's another (likely hardware) issue with your device.
What are you using?

**Reply by samturner3** [reply]
lanbon l8

Replacing

```
"gpio": {
        "config": [
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0
        ]
    },
```
with 
    
```
    "gpio": {
        "config": [
            197658,
            263456,
            329249,
            655628,
            655886,
            656155,
            0,
            0
        ]
    },
```
(Taken config from https://www.openhasp.com/0.6.3/devices/lanbon-l8/) was as MQTT ("_To configure the GPIOs as light switches at once for L8-HS send to topic hasp/<nodename>/config/gpio a message with payload:_") Assuming this should work by placing in file though also?

And save does not persist past reboot, not does it update the gpio config in the settings. 
    
Other things such as pages persists fine.

**Reply by nagyrobi** [reply]
Is that a Dimmer (AU) version?

**Reply by samturner3** [reply]
The hardware I am using is AU however I was given it without the AC back part so not sure if dimmer or not, I have an AC back part for 3 light switches.

I was just testing the file save so the actual config is just the example from the website.

**Reply by nagyrobi** [reply]
:-) Because I experienced exactly the same using the AU dimmer version attached to a 3-gang relay back part. If I configure the GPIOs accordingly, it won't save the files anymore. If I configure it as an AU dimmer (and with the dimmer back part connected) it works.

**Reply by nagyrobi** [reply]
Maybe there's an internal GPIO conflict in the dimmer configured as 3gang and the flash module...?
Thinking loud...

**Reply by fvanroie** [reply]
If you edit the `config.json` file manually those changes will be overwritten when you press `Reboot`. That is because the current settings are saved to flash before a reboot, hence your changes will never be applied/loaded...

If you want to edit the `config.json` file via the File Editor then you should press `Ctrl-S` and do a `Power Cycle` **not** a `Reboot`.
This will prevent the current settings from being saved and overwrite your changes.

The *normal* way would be to use mqtt, serial or telnet console to issue the `config/gpio` command with the JSON string.

**Comment by Sonusss** [comment]
Well, in fact editing manually the config.json is not a solution, it will screw up the unit as all passwords will be erased because they are not appearing as clear text in the file (this is what I get with version 0.6.2).  I suggest to have a kind of "reload" button as for pages to take into consideration modifications made to config.json without touching passwords unless written in clear.
For example in my case I need to edit the config.json because I have several Lanbon L8 in the same room and like them to have exactly the same color (Hue) and I didn't found any other way to set this precisely with the web UI.
